metrics/release_test.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/prometheus/client_golang/prometheus/testutil"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var _ = Describe("Metrics Release", Ordered, func() {
+
+	var (
+		AttemptRunningSecondsHeader = inputHeader{
+			Name: "release_attempt_running_seconds",
+			Help: "Release durations from the moment the release resource was created til the release is marked as running",
+		}
+		AttemptDurationSecondsHeader = inputHeader{
+			Name: "release_attempt_duration_seconds",
+			Help: "Release durations from the moment the release PipelineRun was created til the release is marked as finished",
+		}
+		AttemptTotalHeader = inputHeader{
+			Name: "release_attempt_total",
+			Help: "Total number of releases processed by the operator",
+		}
+	)
+
+	const (
+		validReleaseReason   = "valid_release_reason"
```

Comment:
You can remove these variables and user `v1alpha1.ReleaseReason...` variables.

Reply:
Short comment about this one:
```
package github.com/redhat-appstudio/release-service/metrics
	imports github.com/redhat-appstudio/release-service/api/v1alpha1: import cycle not allowed in test
FAIL	github.com/redhat-appstudio/release-service/metrics [setup failed]
FAIL
```
Metrics is imported by v1alpha1. I kept the old way so far. If you have suggestions, pls come!

Reply:
Right... I actually don't like this but I'm not sure we want to move those constants away from the type itself (which would be the obvious solution). Let's keep it as it is for now.

Reply:
ack